huaweicloud/services/nat/data_source_huaweicloud_nat_private_snat_rules.go

```diff
@@ -0,0 +1,254 @@
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the subnet to which the private SNAT rule belongs.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The description of the private SNAT rule.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the private SNAT rule.",
+			},
+			"transit_ip_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The ID of the transit IP associated with the private SNAT rule.",
+			},
+			"transit_ip_address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The IP address of the transit IP associated with the private SNAT rule.",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The creation time of the private SNAT rule.",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The latest update time of the private SNAT rule.",
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the enterprise project to which the private SNAT rule belongs.",
+			},
+		},
+	}
+	return &sc
+}
+
+func dataSourcePrivateSnatRulesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// listSnatRules: Query the private SNAT rule list
+	var (
+		listSnatRulesHttpUrl = "v3/{project_id}/private-nat/snat-rules"
+		listSnatRulesProduct = "nat"
+	)
+	listSnatRulesClient, err := cfg.NewServiceClient(listSnatRulesProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating NAT client: %s", err)
+	}
+
+	listSnatRulesPath := listSnatRulesClient.Endpoint + listSnatRulesHttpUrl
+	listSnatRulesPath = strings.ReplaceAll(listSnatRulesPath, "{project_id}", listSnatRulesClient.ProjectID)
+
+	listSnatRulesQueryParams := buildListSnatRulesQueryParams(d, cfg)
+	listSnatRulesPath += listSnatRulesQueryParams
+
+	listSnatRulesResp, err := pagination.ListAllItems(
+		listSnatRulesClient,
+		"marker",
+		listSnatRulesPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving private SNAT rule")
+	}
+
+	listSnatRulesRespJson, err := json.Marshal(listSnatRulesResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var listSnatRulesRespBody interface{}
+	err = json.Unmarshal(listSnatRulesRespJson, &listSnatRulesRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("RWQ the Resp is %v", listSnatRulesRespBody)
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(uuid)
+
+	log.Printf("RWQ the value is %v", flattenListSnatRuleResponseBody(listSnatRulesRespBody))
```

Comment:
```suggestion
```
remove it or update the message content
```suggestion
	log.Printf("[DEBUG] The query result of the SNAT rules is %v", flattenListSnatRuleResponseBody(listSnatRulesRespBody))
```

Reply:
done